x/finality/keeper/msg_server.go

```diff
@@ -237,6 +237,33 @@ func (ms msgServer) CommitPubRandList(goCtx context.Context, req *types.MsgCommi
 	return &types.MsgCommitPubRandListResponse{}, nil
 }
 
+func (k Keeper) UnjailFinalityProvider(ctx context.Context, req *types.MsgUnjailFinalityProvider) (*types.MsgUnjailFinalityProviderResponse, error) {
+	defer telemetry.ModuleMeasureSince(types.ModuleName, time.Now(), types.MetricsKeyUnjailFinalityProvider)
+
+	fpPk := req.FpBtcPk
+	info, err := k.FinalityProviderSigningTracker.Get(ctx, fpPk.MustMarshal())
```

Comment:
this fpPk comes from the request, should we validate it first, before call `MustMarshal`?

Reply:
fpPk is already a `BIP340PubKey` type. Why do we need to validate it?